.github/workflows/publish-package.yml

```diff
@@ -84,6 +84,7 @@ jobs:
       - name: Bump version.py
         run: |
           export module=$(sed 's/^.*\///' <<< ${{ matrix.package.name }} | tr '-' '/')
+          export version=$(sed -E 's|-(\d*)|${{  github.ref == 'refs/heads/main' && '' || '-dev' }}\1|g' <<< ${{ matrix.package.version }})
```

Comment:
`pnpm ci:version --snapshot dev` already produces `x.y.z-dev-TIMESTAMP`, there's no need to add another `-dev` to it, just trim down the last dash.

Reply:
weirdly enough I tried that and then the output when running the command locally is wrong. So I had to add it in order to get the correct output of `x.y.z-devTIMESTAMP` for pypi

Reply:
ah, thanks! will do

Reply:
pushed the change, it looks a bit ugly as I read that nested expressions aren't supported. If you were thinking of a more elegant solution feel free to commit it to this branch!

Reply:
i think this one is clearer: `'s|-\([0-9]*\)$|\1|'` (take dash-numbers-eol and turn it into numbers-eol)
also note that sed regexes require `\(` `\)` for grouping, with literal backslashes

Reply:
yeah, that seems cleaner! 
the `-E` argument allows for braces not to be escaped, but can go with the regular version of escaping it.

Reply:
Can pypi recognize `v0.1.0-dev` too? I've only used `v0.1.0.dev0`

Reply:
https://peps.python.org/pep-0440/
> Epoch segment: N!
> Release segment: N(.N)*
> Pre-release segment: {a|b|rc}N
> Post-release segment: .postN
> Development release segment: .devN
>

Reply:
the timestamp is only numbers, so even though it doesn't follow the "spirit" of the spec, i think it technically abides by it. version numbers aren't consecutive but they are ordered